src/pywrangler/pandas/wranglers/interval_identifier.py

```diff
@@ -187,22 +187,30 @@ def _transform(self, series: pd.Series) -> List[int]:
 
         """
 
-        # get boolean series with start and end markers
-        bool_start = series.eq(self.marker_start)
-        bool_end = series.eq(self.marker_end)
+        if self._naive_algorithm:
+            # when series equals marker_column return True
+            bool_start = series.eq(self.marker_start)
```

Comment:
Same line of code appears in line 199 - DRY. There should be need to indent the entire code block with if/else. Instead, the `if self._naive_algorithm:` could be used right after line 199 and return the result `return bool_start.cumsum()` right away. No need for `else` when `return` already exits the function.